File: selection_sort.js
function findSmallest(arr) {
  var smallest = arr[0];
  var smallestIndex = 0;
  //   for (const val of arr) {
  //     if (val < smallest) {
  //       smallest = val;
  //       smallestIndex = arr.findIndex((v) => v === val);
  //     }
  //   }
  for (var i = 1; i < arr.length; i++) {
    if (arr[i] < smallest) {
      smallest = arr[i];
      smallestIndex = i;
    }
  }
  return smallestIndex;
}

function selectionSort(arr) {
  var newArr = [];
  var len = arr.length;
  for (var i = 0; i < len; i++) {
    var smallestIndex = findSmallest(arr);
    newArr.push(arr.splice(smallestIndex, 1)[0]);
  }
  return newArr;
}

console.log(selectionSort([5, 3, 6, 2, 10]));

File: selection_sort.py
# 找出数组中最小数
def findSmallest(arr):
    smallest = arr[0]
    smallestIndex = 0
    for i in range(1, len(arr)):
        if arr[i] < smallest:
            smallest = arr[i]
            smallestIndex = i
    return smallestIndex

# 选择排序
def selectionSort(arr):
    newArr = []
    for i in range(len(arr)):
        # 找出数组中最小数，并加入到新数组中
        smallestIndex = findSmallest(arr)
        newArr.append(arr.pop(smallestIndex))
    return newArr


print(selectionSort([5, 3, 6, 2, 10]))
